ui-react/packages/atlasmap/src/UI/ConditionalExpressionInput.tsx

```diff
@@ -1,30 +1,112 @@
-import React, { FunctionComponent } from "react";
-import { ToolbarGroup, ToolbarItem, Button } from "@patternfly/react-core";
-import { css, StyleSheet } from "@patternfly/react-styles";
+import React, { FunctionComponent, useState } from "react";
 import {
-  // ExpressionContent,
-  IExpressionContentProps,
-} from "./ExpressionContent";
+  ToolbarGroup,
+  ToolbarItem,
+  Button,
+  Tooltip,
+} from "@patternfly/react-core";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { ExpressionContent } from "./ExpressionContent";
 
 const styles = StyleSheet.create({
   toolbarItem: { flex: 1 },
 });
+export interface IConditionalExpressionInputProps {
+  executeFieldSearch: (searchFilter: string, isSource: boolean) => string[][];
+  mappingExpressionAddField: (
+    selectedField: string,
+    newTextNode: any,
+    atIndex: number,
+    isTrailer: boolean,
+  ) => void;
+  mappingExpressionClearText: (
+    nodeId?: string,
+    startOffset?: number,
+    endOffset?: number,
+  ) => any;
+  isMappingExpressionEmpty: boolean;
+  mappingExpressionInit: () => void;
+  mappingExpressionInsertText: (
+    str: string,
+    nodeId?: string | undefined,
+    offset?: number | undefined,
+  ) => void;
+  mappingExpressionObservable: () => any;
+  mappingExpressionRemoveField: (
+    tokenPosition?: string,
+    offset?: number,
+    removeNext?: boolean,
+  ) => void;
+  onConditionalMappingExpressionEnabled: () => boolean;
+  onToggleExpressionMode: () => void;
+  mappingExpression?: string;
+  trailerId: string;
+}
+
+export const ConditionalExpressionInput: FunctionComponent<IConditionalExpressionInputProps> = ({
+  executeFieldSearch,
+  mappingExpressionAddField,
+  mappingExpressionClearText,
+  isMappingExpressionEmpty,
+  mappingExpressionInit,
+  mappingExpressionInsertText,
+  mappingExpressionObservable,
+  mappingExpressionRemoveField,
+  mappingExpression,
+  onConditionalMappingExpressionEnabled,
+  onToggleExpressionMode,
+  trailerId,
+}) => {
+  const [conditionExpressionEnabled, setCondExprEnabled] = useState<boolean>(
```

Comment:
If I follow the trail I see that the callback will eventually [set this flag](https://github.com/atlasmap/atlasmap/blob/ea7f43e4abec4684cc83718102b6b17af16adada/ui-react/packages/atlasmap-core/src/services/mapping-management.service.ts#L592). With this useState we are creating a copy of that flag. Since we want to have a single source of truth and since the expression box engine is provided by the RxJS code, we should make that flag available through the `AtlasmapProvider` and feed its value to this component.

Reply:
is this issue significant enough to block merging or can this be addressed is a separate issue?  There are a few things left to be fixed (d&d, caret positioning, etc).  I'd like to get the bulk checked in before it goes stale.

Reply:
Ok - I've addressed this issue with the latest change.